tests/two_queue_tests.cpp

```diff
@@ -0,0 +1,211 @@
+   app->executor().post( priority::high,   app->executor().queue2(), [&]() { rslts[9]=seq_num; ++seq_num; } );
+
+   // Stop application. Use lowest at the end to make sure this executes the last
+   app->executor().post( priority::lowest, app->executor().queue1(), [&]() {
+      // queue1 should have current function and queue2 should have all its functions
+      BOOST_REQUIRE_EQUAL( app->executor().queue1().size(), 1);
+      BOOST_REQUIRE_EQUAL( app->executor().queue2().size(), 10 );
+      app->quit();
+      } );
+   app_thread.join();
+
+   // both queues are cleared after execution 
+   BOOST_REQUIRE_EQUAL( app->executor().queue1().empty(), true);
+   BOOST_REQUIRE_EQUAL( app->executor().queue2().empty(), true);
+
+   // no results
+   BOOST_REQUIRE_EQUAL( rslts.size(), 0 );
+}
+
+// verify functions from both queues are processed when mode is set to both_queues
+BOOST_AUTO_TEST_CASE( execute_from_both_queues ) {
+   appbase::scoped_app app;
+   auto app_thread = start_app_thread(app);
+
+
+   // set to run functions from both queues
+   app->executor().set_exec_mode(appbase::exec_mode::both_queues);
+
+   // post functions
+   std::map<int, int> rslts {};
+   int seq_num = 0;
+   app->executor().post( priority::medium, app->executor().queue1(), [&]() { rslts[0]=seq_num; ++seq_num; } );
+   app->executor().post( priority::medium, app->executor().queue2(), [&]() { rslts[1]=seq_num; ++seq_num; } );
+   app->executor().post( priority::high,   app->executor().queue2(), [&]() { rslts[2]=seq_num; ++seq_num; } );
+   app->executor().post( priority::lowest, app->executor().queue1(), [&]() { rslts[3]=seq_num; ++seq_num; } );
+   app->executor().post( priority::low,    app->executor().queue1(), [&]() { rslts[4]=seq_num; ++seq_num; } );
+   app->executor().post( priority::low,    app->executor().queue2(), [&]() { rslts[5]=seq_num; ++seq_num; } );
+   app->executor().post( priority::highest,app->executor().queue1(), [&]() { rslts[6]=seq_num; ++seq_num; } );
+   app->executor().post( priority::low,    app->executor().queue2(), [&]() { rslts[7]=seq_num; ++seq_num; } );
+   app->executor().post( priority::lowest, app->executor().queue1(), [&]() { rslts[8]=seq_num; ++seq_num; } );
+   app->executor().post( priority::lowest, app->executor().queue1(), [&]() { rslts[9]=seq_num; ++seq_num; } );
+   app->executor().post( priority::low,    app->executor().queue2(), [&]() { rslts[10]=seq_num; ++seq_num; } );
+   app->executor().post( priority::medium, app->executor().queue2(), [&]() { rslts[11]=seq_num; ++seq_num; } );
+
+   // stop application. Use lowest at the end to make sure this executes the last
+   app->executor().post( priority::lowest, app->executor().queue1(), [&]() {
+      // queue1 should have current function and queue2's functions are all executed 
+      BOOST_REQUIRE_EQUAL( app->executor().queue1().size(), 1);
+      BOOST_REQUIRE_EQUAL( app->executor().queue2().size(), 0 );
```

Comment:
Same comment as above.

Reply:
These comments show why the two queue implementation should *not* be in appbase. It is specific to leap and unnecessarily add complexity to appbase.